src/audio/base_fw_intel.c

```diff
@@ -183,6 +183,10 @@ __cold static int basefw_mem_state_info(uint32_t *data_offset, char *data)
 	size = ALIGN(size, 4);
 	/* size is also saved as tuple length */
 	tuple_data = rballoc(SOF_MEM_FLAG_USER, size);
+	if (!tuple_data) {
+		LOG_ERR("basefw_mem_state_info(): allocation failed");
+		return IPC4_ERROR_INVALID_PARAM;
```

Comment:
I think there's an IPC4 error for "out of memory"

Reply:
I initially thought that too, but no.

Reply:
@lgirdwood IPC4_OUT_OF_MEMORY

Reply:
@lgirdwood should we update this one while fixing `rfree()`?